src/stores/store.test.ts

```diff
@@ -10,4 +11,20 @@ describe('store actions', () => {
     });
     expect(result.current.currentView).toBe('upload');
   });
+
+  it('should set dataTable', () => {
+    const { result } = renderHook(() => useStore());
+    act(() => {
+      result.current.setDataTable(mockDataTable);
+    });
+    expect(result.current.dataTable).toEqual(mockDataTable);
+  });
+
+  it('should initialize columns', () => {
+    const { result } = renderHook(() => useStore());
+    act(() => {
+      result.current.initializeColumns(mockDataTable);
+    });
+    expect(result.current.columns).toEqual(mockDataTable);
+  });
```

Comment:
shouldn't that read `mockHeaders` instead? It would be a lot clearer here if you put the expected array of columns explicitly in the test like so
```suggestion
      result.current.initializeColumns(mockDataTable);
    });
    expect(result.current.columns).toEqual(
        [
            'participant_id', 
            'age',
            'sex'
        ]
    );
  });
```